webapp/src/helpers/greeting.ts

```diff
@@ -0,0 +1,151 @@
+  ltr: 'ltr',
+});
+morning_greetings.set('fr', {
+  beginning_punct: '',
+  greeting: 'Bon matin',
+  delimiter: ', ',
+  ending_punct: '!',
+  ltr: 'ltr',
+});
+morning_greetings.set('ja', {
+  beginning_punct: '',
+  greeting: 'おはようございます',
+  delimiter: '、',
+  ending_punct: '!',
+  ltr: 'ltr',
+});
+
+let afternoon_greetings = new Map<string, GreetingType>();
+afternoon_greetings.set('en', {
+  beginning_punct: '',
+  greeting: 'Good afternoon',
+  delimiter: ', ',
+  ending_punct: '!',
+  ltr: 'ltr',
+});
+afternoon_greetings.set('fr', {
+  beginning_punct: '',
+  greeting: 'Bonne après-midi',
+  delimiter: ', ',
+  ending_punct: '!',
+  ltr: 'ltr',
+});
+afternoon_greetings.set('ja', {
+  beginning_punct: '',
+  greeting: 'こんにちは',
+  delimiter: '、',
+  ending_punct: '!',
+  ltr: 'ltr',
+});
+
+let evening_greetings = new Map<string, GreetingType>();
+evening_greetings.set('en', {
+  beginning_punct: '',
+  greeting: 'Good evening',
+  delimiter: ', ',
+  ending_punct: '!',
+  ltr: 'ltr',
+});
+evening_greetings.set('fr', {
+  beginning_punct: '',
+  greeting: 'Bonsoir',
+  delimiter: ', ',
+  ending_punct: '!',
+  ltr: 'ltr',
+});
+evening_greetings.set('ja', {
+  beginning_punct: '',
+  greeting: 'こんばんは',
+  delimiter: '、',
+  ending_punct: '!',
+  ltr: 'ltr',
+});
+
+/**
+ * Returns the greeting string based on the different time of the day in a preferred language.
+ *
+ * @param timeOfDay - "morning", "afternoon" or "evening" depending on the time of the day
+ * @param preferredLanguage - the browser language two-character code
+ * @param username - user's name
+ * @returns The user_greeting_string which includes the structure of the greeting string (ex. "Good morning, {username}!")
+ *
+ */
+const getPreferredLanguageGreeting = (
+  timeOfDay: string,
+  preferredLanguage: string,
+  username: string
+): string => {
+  let user_greeting: GreetingType | any = {};
+
+  if (timeOfDay === 'morning') {
+    user_greeting =
+      morning_greetings.get(preferredLanguage) || morning_greetings.get('en')!;
+  } else if (timeOfDay === 'afternoon') {
+    user_greeting =
+      afternoon_greetings.get(preferredLanguage) ||
+      afternoon_greetings.get('en')!;
+  } else if (timeOfDay === 'evening') {
+    user_greeting =
+      evening_greetings.get(preferredLanguage) || evening_greetings.get('en')!;
+  }
+
+  const user_greeting_string =
+    user_greeting.beginning_punct +
+    user_greeting.greeting +
+    user_greeting.delimiter +
+    username +
+    user_greeting.ending_punct;
+
+  return user_greeting_string;
+};
+
+/**
+ * Gets hour based on Date() and returns the greeting string based on the different time of the day.
+ *
+ * @param username - user's name
+ * @returns The string which includes the icon and the greeting string depending on the time of the day (ex. "🌅 Good morning, {username}!")
+ *
+ */
+export const getGreeting = (username: string): string => {
+  const myDate = new Date();
+  const hrs = myDate.getHours();
```

Comment:
If you're only using `myDate` once here, can you do `const hrs = new Date().getHours()` instead? Or are you planning to extract more data from `myDate` later?

Reply:
We don't use `myDate` anywhere else, so as you said, I can chain `.getHours()` to `Date()`.